gateway/api/views.py

```diff
@@ -453,6 +453,18 @@ def retrieve(self, request, pk=None):  # pylint: disable=unused-argument
         tracer = trace.get_tracer("gateway.tracer")
         ctx = TraceContextTextMapPropagator().extract(carrier=request.headers)
         with tracer.start_as_current_span("gateway.job.retrieve", context=ctx):
+            job = Job.objects.filter(pk=pk).first()
+            if job is None:
+                logger.warning("Job [%s] not found", pk)
+                return Response(status=404)
```

Comment:
I think I would add here a message as we have in other places like [here](https://github.com/Qiskit/qiskit-serverless/blob/main/gateway/api/views.py#L221). And I would try to use official status from django like [here](https://github.com/Qiskit/qiskit-serverless/blob/main/gateway/api/views.py#L222)